### PR TITLE
DEVPROD-761 Specify backport commit-sha option only allows from same github project

### PR DESCRIPTION
**Pull request by ZackarySantana**
DEVPROD-761

### Description
A user was attempting to specify commit-sha for a different repository than the project they were using it on, which caused the sha not to be found

### CODE REVIEW

**Review by ablack12 — CHANGES_REQUESTED** [review]
Can we also add something to the output for `--commit-sha` from the help text for the backport command? 
https://github.com/evergreen-ci/evergreen/blob/f29c49c4c4336df43df5dd80e8774c5ce0cc1d8e/operations/commit_queue.go#L369

**Comment by ZackarySantana** [comment]
> Can we also add something to the output for `--commit-sha` from the help text for the backport command?
> 
> https://github.com/evergreen-ci/evergreen/blob/f29c49c4c4336df43df5dd80e8774c5ce0cc1d8e/operations/commit_queue.go#L369

I added it to commit sha and the patch alias flags to the command. I was thinking of adding that they are mutually exclusive as well like we state in the docs, but I wasn't sure how to word it and wasn't sure if someone would try it + if they did it would error.

File: operations/commit_queue.go [hunk 1]
```diff
@@ -362,11 +362,11 @@ func backport() cli.Command {
 				},
 				cli.StringFlag{
 					Name:  joinFlagNames(existingPatchFlag, "e"),
-					Usage: "existing commit queue patch",
+					Usage: "existing commit queue patch from another repository",
```
Comment (ablack12) [inline operations/commit_queue.go:365]: I think this isn't likely to be a different repository but a different _project_ (or a different branch) bc backports happen to different branches on the same repo

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: operations/commit_queue.go (hunk 1 above)
Comment (ZackarySantana) [reply operations/commit_queue.go:365]: Oh! That was my misunderstanding, let me rephrase that

**Review by ablack12 — APPROVED** [review]
